third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -2645,6 +2659,7 @@ struct StructDef {
     field_count: u16,
     // `ModuelCache::structs` global table index
     idx: CachedStructIndex,
+    definition_struct_type: Arc<StructType>,
```

Comment:
Yes! But was curious what's the benefit of using `Weak` other than it is indeed `Weak`?

Reply:
It's cheaper (doesn't need to touch the strong count) and can also document that this is like a reference, i.e. an Arc is expected to exist somewhere, although it can be used otherwise as well (try_upgrade) - in Rust documentation they mention it primarily for breaking circular dependencies (if Arcs form a circular dependency they can never get deallocated because they will keep each other alive).
Maybe not worth optimizing here and now, let's revisit this later.